### PR TITLE
Adding frontend-base

### PR DESCRIPTION
**Pull request by davidjoy**
It'll be easiest to review this commit by commit.  They're pretty orderly.

### CODE REVIEW

File: src/account-settings/AccountSettingsPage.jsx [hunk 1]
```diff
@@ -53,11 +54,29 @@ class AccountSettingsPage extends React.Component {
...
+    this.state = {
+      duplicateTpaProvider,
+    };
   }
 
   componentDidMount() {
+    this.props.fetchUserAccount(this.context.authenticatedUser.username);
     this.props.fetchSettings();
     this.props.fetchSiteLanguages();
+    sendTrackingLogEvent('edx.user.settings.viewed', {
```
Comment (abutterworth) [inline src/account-settings/AccountSettingsPage.jsx]: Is this true? We've just sent the request to fetch the user account so maybe the page is still blank?

File: src/account-settings/data/service.js [hunk 2]
```diff
@@ -1,41 +1,17 @@
+import { App } from '@edx/frontend-base';
 import pick from 'lodash.pick';
 import omit from 'lodash.omit';
 import isEmpty from 'lodash.isempty';
 
-import { applyConfiguration, handleRequestError, unpackFieldErrors } from '../common/serviceUtils';
-import { configureService as configureDeleteAccountApiService } from './delete-account';
-import { configureService as configureResetPasswordApiService } from './reset-password';
-import { configureService as configureSiteLanguageApiService } from './site-language';
-import { configureService as configureThirdPartyAuthApiService, getThirdPartyAuthProviders } from './third-party-auth';
```
Comment (abutterworth) [inline src/account-settings/data/service.js]: Great to remove so many of these.

File: src/account-settings/delete-account/data/sagas.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ import {
   deleteAccountFailure,
 } from './actions';
 
-import { postDeleteAccount } from './service';
+import postDeleteAccount from './service';
```
Comment (abutterworth) [inline src/account-settings/delete-account/data/sagas.js]: I think we should ignore the prefer default export rule here and keep the named export. `postDeleteAccount` from `service.js` seems odd at first read.

File: src/account-settings/AccountSettingsPage.jsx (hunk 1 above)
Comment (davidjoy) [reply src/account-settings/AccountSettingsPage.jsx]: I moved this code from index.jsx, where it was being done even earlier.  So this seems more true?  I'm a bit disinclined to change the behavior too much as part of this PR, fwiw.

File: src/account-settings/data/service.js (hunk 2 above)
Comment (davidjoy) [reply src/account-settings/data/service.js]: Yeah, that was satisfying. :)

File: src/account-settings/delete-account/data/sagas.js (hunk 3 above)
Comment (davidjoy) [reply src/account-settings/delete-account/data/sagas.js]: Yeah... I think you're right.  Going to fix that.

**Pull request merged**